srv/ListJobs.srv

```diff
@@ -0,0 +1,22 @@
+# SPDX-FileCopyrightText: 2024, Yaskawa America, Inc.
+# SPDX-FileCopyrightText: 2024, Delft University of Technology
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# empty request
+
+---
+
+# Result of the service invocation. Values other than one (1) signal failure.
+#
+# NOTE: future versions of this service may use a different set of result codes
+uint32 result_code
+
+# string representation of the value in 'result_code', for humans
+string message
+
+# The list of jobs present on the controller.
+#
+# NOTE: these are job names, not filenames, and thus will not include the file
+# extension (JBI)
+string[] names
```

Comment:
Seeing as we should support non-ASCII configurations as well: ROS 2 [does support unicode/wstring](https://docs.ros.org/en/humble/Concepts/Basic/About-Interfaces.html#field-types), but I'm not sure about the exact status of that support.

Additionally: since there is an upper limit to the length of job names, we could add it here. I didn't do that because I don't know whether we'd like to keep this compatible with future controllers which may not have such limitations.

Reply:
> Seeing as we should support non-ASCII configurations as well

Traditionally, we don't worry about that when transmitting a job name. For other SDK's, we just send it as ascii and it's up to the user to figure out the encoding.

Obviously this isn't the most user-friendly. But we don't have any encoding information in the job file. And the display language of the pendant doesn't always match what the job was created in. (Semi-common with multi-national companies.)

> Additionally: since there is an upper limit to the length of job names, we could add it here. 

Technically, the limit is on the number of bytes and not the number of characters. So for a japanese job, the limit is 16 characters.

So I'd like to just let the pendant enforce any limits

Reply:
Ok. I just wanted to draw attention to these points.

I'll add a short explanation as to why we don't use a `wstring` for the `names` field, probably paraphrasing your comment here.